### PR TITLE
Update dependencies, fix lints and set updated packages

### PR DESCRIPTION
**Pull request by saraswatpuneet**
## Details
- [x] Retain current version of polkadotjs, multiformats 
- [x] Remove unused dependencies
- [x] Update linter and fix all the linter related errors, most of them were `namespace` errors 
- [x] Update remaning depdencies to new version

## Tests
- [x] Run e2e tests to ensure everything is working

### CODE REVIEW

File: apps/worker/src/batch_announcer/batch.announcer.ts [hunk 1]
```diff
@@ -50,7 +50,7 @@ export class BatchAnnouncer {
     const writer = await ParquetWriter.openStream(parquetSchema, publishStream as any, writerOptions);
     // eslint-disable-next-line no-restricted-syntax
     for await (const announcement of announcements) {
-      await writer.appendRow(announcement);
+      await writer.appendRow(announcement as Record<string, any>);
```
Comment (saraswatpuneet) [inline apps/worker/src/batch_announcer/batch.announcer.ts:53]: No clue why after updates, this error started popping, got fixed by a cast but not sure if its correct

```

> ERROR in ./apps/worker/src/batch_announcer/batch.announcer.ts:53:30
> TS2345: Argument of type 'Announcement' is not assignable to parameter of type 'Record<string, unknown>'.
>   Type '{ announcementType: AnnouncementType; fromId: string; } & TombstoneFields' is not assignable to type 'Record<string, unknown>'.
>     Index signature for type 'string' is missing in type '{ announcementType: AnnouncementType; fromId: string; } & TombstoneFields'.
>     51 |     // eslint-disable-next-line no-restricted-syntax
>     52 |     for await (const announcement of announcements) {
>   > 53 |       await writer.appendRow(announcement);
>        |                              ^^^^^^^^^^^^
>     54 |     }
>     55 |     await writer.close();
>     56 |
> 
> webpack 5.90.1 compiled with 1 error in 13268 ms
> 

```

File: apps/api/src/api.module.ts [hunk 2]
```diff
@@ -2,14 +2,14 @@ import { Module } from '@nestjs/common';
...
-import { RedisModule } from '@liaoliaots/nestjs-redis';
+import { RedisModule } from '@songkeys/nestjs-redis';
 import { BullBoardModule } from '@bull-board/nestjs';
 import { BullMQAdapter } from '@bull-board/api/bullMQAdapter';
 import { ExpressAdapter } from '@bull-board/express';
 import { MulterModule } from '@nestjs/platform-express';
 import { ApiController } from './api.controller';
 import { DevelopmentController } from './development.controller';
-import { QueueConstants } from '../../../libs/common/src';
+import { ASSET_QUEUE_NAME, BATCH_QUEUE_NAME, BROADCAST_QUEUE_NAME, PROFILE_QUEUE_NAME, PUBLISH_QUEUE_NAME, REACTION_QUEUE_NAME, REPLY_QUEUE_NAME, REQUEST_QUEUE_NAME, STATUS_QUEUE_NAME, TOMBSTONE_QUEUE_NAME, TRANSACTION_RECEIPT_QUEUE_NAME, UPDATE_QUEUE_NAME } from '../../../libs/common/src';
```
Comment (saraswatpuneet) [inline apps/api/src/api.module.ts:12]: linter errored about not allowing namespaces anymore, so most of the changes are related to removal of namespace and importing types directly

File: apps/api/src/api.module.ts [hunk 3]
```diff
@@ -2,14 +2,14 @@ import { Module } from '@nestjs/common';
 import { EventEmitterModule } from '@nestjs/event-emitter';
 import { BullModule } from '@nestjs/bullmq';
 import { ScheduleModule } from '@nestjs/schedule';
-import { RedisModule } from '@liaoliaots/nestjs-redis';
+import { RedisModule } from '@songkeys/nestjs-redis';
```
Comment (saraswatpuneet) [inline apps/api/src/api.module.ts:5]: This was erroring as the other flavor of redis is not upto date, changing to this similar one compatible with newer nestjs, semantics around redis remains the same

File: apps/api/src/api.module.ts (hunk 2 above)
Comment (wilwade) [reply apps/api/src/api.module.ts:12]: If you want this change to be smaller you can do this instead on these:

```suggestion
import * as QueueConstants from '../../../libs/common/src';
```

File: package.json [hunk 4]
```diff
@@ -42,83 +42,66 @@
...
     "@dsnp/activity-content": "^1.1.0",
-    "@dsnp/frequency-schemas": "^1.0.2",
-    "@dsnp/parquetjs": "^1.3.4",
-    "@frequency-chain/api-augment": "1.7.0",
-    "@jest/globals": "^29.5.0",
-    "@liaoliaots/nestjs-redis": "^9.0.5",
+    "@dsnp/frequency-schemas": "^1.1.0",
+    "@dsnp/parquetjs": "^1.6.2",
+    "@frequency-chain/api-augment": "1.11.1",
+    "@jest/globals": "^29.7.0",
```
Comment (wilwade) [inline package.json:53]: This was in here before, but jest stuff should be a devDep

File: package.json [hunk 5]
```diff
@@ -42,83 +42,66 @@
...
+    "@nestjs/bullmq": "^10.1.1",
+    "@nestjs/cli": "^10.3.2",
+    "@nestjs/common": "^10.3.8",
+    "@nestjs/config": "^3.2.2",
+    "@nestjs/core": "^10.3.8",
+    "@nestjs/event-emitter": "^2.0.4",
+    "@nestjs/platform-express": "^10.3.8",
+    "@nestjs/schedule": "^4.0.2",
+    "@nestjs/swagger": "^7.3.1",
+    "@nestjs/testing": "^10.3.8",
```
Comment (wilwade) [inline package.json:64]: Move to devDeps?

File: eslint.config.mjs [hunk 6]
```diff
@@ -0,0 +1,26 @@
+// @ts-check
+
+import eslint from '@eslint/js';
```
Comment (wilwade) [inline eslint.config.mjs:3]: I don't see this in the devDeps, we might need to add it

File: apps/worker/src/request_processor/request.processor.service.ts [hunk 7]
```diff
@@ -49,7 +49,7 @@ export class RequestProcessorService extends BaseConsumer {
       // exponential backoff
       const delayedTime = 2 ** data.dependencyAttempt * this.configService.assetUploadVerificationDelaySeconds * MILLISECONDS_PER_SECOND;
       await job.moveToDelayed(Date.now() + delayedTime, job.token);
-      await job.update(data);
+      await job.updateData(data);
```
Comment (wilwade) [inline apps/worker/src/request_processor/request.processor.service.ts:52]: This change surprised me.

Comment (saraswatpuneet) [reply apps/worker/src/request_processor/request.processor.service.ts:52]: Yep this function is not there in this new redis replacement, maybe dev changed the name to be more appropriate

File: eslint.config.mjs (hunk 6 above)
Comment (saraswatpuneet) [reply eslint.config.mjs:3]: I think this is packaged with eslint but will check

File: apps/worker/src/batch_announcer/batch.announcer.ts (hunk 1 above)
Comment (wilwade) [reply apps/worker/src/batch_announcer/batch.announcer.ts:53]: I checked and this is an issue with Parquetjs. I'll work on getting a fix in over there.

**Review by wilwade — APPROVED** [review]
🙇

File: apps/worker/src/batch_announcer/batch.announcer.ts (hunk 1 above)
Comment (wilwade) [reply apps/worker/src/batch_announcer/batch.announcer.ts:53]: @saraswatpuneet I found the correct fix and it isn't in the Parquetjs side.

The TypedAnnouncement type needs to explicitly ~extend~ intersect with the `Record<string, unknown>`
image

**Pull request merged**